presto-main/src/main/java/io/prestosql/operator/scalar/DateTimeFunctions.java

```diff
@@ -139,6 +144,28 @@ public static long fromUnixTime(@SqlType(StandardTypes.DOUBLE) double unixTime,
         return packDateTimeWithZone(Math.round(unixTime * 1000), zoneId.toStringUtf8());
     }
 
+    @ScalarFunction("from_unixtime_nanos")
+    @SqlType("timestamp(9)")
+    public static LongTimestamp fromUnixTimeNanos(@SqlType(StandardTypes.BIGINT) long unixTimeNanos)
+    {
+        long epochSeconds = unixTimeNanos / NANOSECONDS_IN_SECOND;
+        long nanosOfSecond = unixTimeNanos % NANOSECONDS_IN_SECOND;
+        long picosOfSecond = 1000L * nanosOfSecond;
+        return DateTimes.longTimestamp(epochSeconds, picosOfSecond);
+    }
+
+    @ScalarFunction("from_unixtime_nanos")
+    @LiteralParameters({"p", "s"})
+    @SqlType("timestamp(9)")
+    public static LongTimestamp fromUnixTimeNanos(@SqlType("decimal(p, s)") Slice arg)
+    {
+        BigInteger unixTimeNanos = Decimals.decodeUnscaledValue(arg);
```

Comment:
This is incorrect because it returns `BigInteger` representing whole "unscaled" value. Not just integral part.
That is for `DECIMAL 123.456` you will get `123456`.
Here we want to look at just integral part (module the fact that we want to add 1 to it if factional part is >= 0.5)